### ISSUE TITLE
Why there is stop_gradient for td?

### ISSUE THREAD

**Issue by mas17kaworu** [MorvanZhou/Reinforcement-learning-with-tensorflow#47]
https://github.com/MorvanZhou/Reinforcement-learning-with-tensorflow/blob/b484df7fc7dadab61e73e04aa40416bf6db68321/contents/10_A3C/A3C_discrete_action.py#L63

In this line, I can't understand why there's stop_gradient before td

为啥要对td做反向截断呢？
我看了些别人的A3C代码，都没有这么做。而且
https://github.com/MorvanZhou/Reinforcement-learning-with-tensorflow/blob/b484df7fc7dadab61e73e04aa40416bf6db68321/experiments/Robot_arm/A3C.py#L82
robot arm中 stop_gradient 消失了。真的很不理解。望赐教。谢谢！

**Comment by MorvanZhou** [comment]
其实我在这犹豫了很久, 我个人理解是:
`self.a_loss = tf.reduce_mean(-log_prob * tf.stop_gradient(td))`, 这 `td` 是 critic, 不是 actor,  `a_loss` 是更新 actor, 如果 gradient 反向传去了 `td`, 就使得 `a_loss` 更新了 Critic, 虽然我在这里使用了`apply_gradient()` 给单独的 variables:
https://github.com/MorvanZhou/Reinforcement-learning-with-tensorflow/blob/b484df7fc7dadab61e73e04aa40416bf6db68321/contents/10_A3C/A3C_discrete_action.py#L78-L79
但是为了保险起见, 我还是加入了 `stop_gradient()`.  现在想想, 有了 apply_gradient to paticular varables, 应该就不用写 stop_gradient了
